typed-racket-lib/typed-racket/base-env/base-special-env.rkt

```diff
@@ -290,12 +290,15 @@
   ;; check-in-port
   [(make-template-identifier 'check-in-port 'racket/private/for)
    (-> Univ Univ Univ)]
-  ;; from the expansion of `with-syntax'
+  ;; from the expansion of `with-syntax' or `quasisyntax'
   [(make-template-identifier 'apply-pattern-substitute 'racket/private/stxcase)
    (->* (list (-Syntax Univ) Univ) Univ (-Syntax Univ))]
   ;; same
   [(make-template-identifier 'with-syntax-fail 'racket/private/with-stx)
    (-> (-Syntax Univ) (Un))]
+  ; from `quasisyntax'
+  [(make-template-identifier 'check-splicing-list 'racket/private/qqstx)
+   (-> Univ (-Syntax Univ) (-Syntax Univ))]
```

Comment:
This could be more precise, re-using the type for `datum->syntax`, but it's fine to leave it alone at the moment.